components/togglebox.tsx

```diff
@@ -0,0 +1,99 @@
+import { ChangeEvent, InputHTMLAttributes } from "react";
+import { AnimatePresence, motion } from "framer-motion";
+import { RemixiconComponentType } from "@remixicon/react";
+import { cn } from "./../lib/utils";
+import LoadingSpinner from "./loading-spinner";
+
+interface ToggleboxProps extends InputHTMLAttributes<HTMLInputElement> {
+  checked?: boolean;
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
+  icon?: RemixiconComponentType | null;
+  isLoading?: boolean;
+  name?: string;
+  label?: string;
+  description?: string;
+  containerClassName?: string;
+  titleClassName?: string;
+}
+
+export default function Togglebox({
+  name,
+  checked = false,
+  onChange,
+  icon: Icon = null,
+  isLoading = false,
+  label,
+  description,
+  containerClassName,
+  titleClassName,
+  ...props
+}: ToggleboxProps) {
+  const inputId = `togglebox-${name}-${props.value}`;
```

Comment:
let's not do this, we can rather accept an optional `id` props, and hand them down to the `input` component correctly, let's trust our user they can set the `id` and we should just pass them as-is without modifying